doc/source/tutorials/contracting_clusters/assets/clusters.py

```diff
@@ -0,0 +1,45 @@
+import igraph as ig
+import matplotlib.pyplot as plt
+import random
+
+# Generate a random geometric graph with random vertex sizes 
+random.seed(0)
+g1 = ig.Graph.GRG(50, 0.15)
+g1.vs['vertex_size'] = [random.uniform(0.03, 0.08) for i in range(g1.vcount())]
+
+# Cluster graph into weakly connected components
+clusters = g1.clusters(mode='weak')
+nclusters = len(clusters)
+
+# Visualise different components
+fig, axs = plt.subplots(1, 2)
+ig.plot(
+    clusters,
+    target=axs[0],
+    palette=ig.RainbowPalette(),
+    vertex_size=g1.vs['vertex_size'],
+    vertex_color=list(map(int, ig.rescale(clusters.membership, (0, 200), clamp=True))),
+    edge_width=0.7
+)
+
+# Contract vertices, using mean value for coordinates, and using the max vertex_size as the representation
+def mean(list):
+    return sum(list) / len(list)
+
+g2 = clusters.cluster_graph(combine_vertices={'x':mean, 'y':mean, 'vertex_size':max})
```

Comment:
`combine_vertices` understands `"mean"` (as a string) and `"max"` (also as a string) so there is no need to define `mean()` on your own. On the other hand, it's good to show that the `combine_vertices=...` kwarg works with arbitrary Python functions. The implementation behind `"mean"` and `"max"` (as strings) is implemented in C. Other options are: `"sum"`, `"prod"`, `"min"`, `"random"`, `"first"`, `"last"`, `"median"` and `"concat"` (for strings).